inbm/dispatcher-agent/dispatcher/source/linux_gpg_key.py

```diff
@@ -61,4 +62,4 @@ def add_gpg_key(remote_key_path: str, key_store_name: str) -> None:
         raise SourceError(f"Error getting GPG key from remote source: {e}")
 
     except subprocess.CalledProcessError as e:
-        raise SourceError(f"Error running gpg command to dearmor key: {e}")
+        raise SourceError(f"Error running GPG command to dearmor key: {e}")
```

Comment:
**Suggestion:** Consider using a context manager to ensure that the subprocess.Popen object is properly cleaned up after use. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    with subprocess.Popen(
        ["/usr/bin/gpg", "--dearmor", "--output", key_path],
        stdin=subprocess.PIPE,
        text=True,
        shell=False,
    ) as proc:
        proc.communicate(input=decoded_ascii)
        if proc.returncode != 0:
            raise SourceError(f"Error running GPG command to dearmor key: return code {proc.returncode}")
```